pandas/tests/resample/test_datetime_index.py

```diff
@@ -25,1484 +23,1491 @@
+    expected = ts.resample('A-JUN').mean()
+    expected.index = period_range('1990', '2000', freq='a-jun')
+    assert_series_equal(result, expected)
+
+    result = ts.resample('M', kind='period').mean()
+    expected = ts.resample('M').mean()
+    expected.index = period_range('1990-01', '2000-01', freq='M')
+    assert_series_equal(result, expected)
+
+    result = ts.resample('M', kind='period').mean()
+    expected = ts.resample('M').mean()
+    expected.index = period_range('1990-01', '2000-01', freq='M')
+    assert_series_equal(result, expected)
+
+
+def test_ohlc_5min():
+    def _ohlc(group):
+        if isna(group).all():
+            return np.repeat(np.nan, 4)
+        return [group[0], group.max(), group.min(), group[-1]]
+
+    rng = date_range('1/1/2000 00:00:00', '1/1/2000 5:59:50', freq='10s')
+    ts = Series(np.random.randn(len(rng)), index=rng)
+
+    resampled = ts.resample('5min', closed='right',
+                            label='right').ohlc()
+
+    assert (resampled.loc['1/1/2000 00:00'] == ts[0]).all()
+
+    exp = _ohlc(ts[1:31])
+    assert (resampled.loc['1/1/2000 00:05'] == exp).all()
+
+    exp = _ohlc(ts['1/1/2000 5:55:01':])
+    assert (resampled.loc['1/1/2000 6:00:00'] == exp).all()
+
+
+def test_downsample_non_unique():
+    rng = date_range('1/1/2000', '2/29/2000')
+    rng2 = rng.repeat(5).values
+    ts = Series(np.random.randn(len(rng2)), index=rng2)
+
+    result = ts.resample('M').mean()
+
+    expected = ts.groupby(lambda x: x.month).mean()
+    assert len(result) == 2
+    assert_almost_equal(result[0], expected[1])
+    assert_almost_equal(result[1], expected[2])
 
-    def test_resample_rounding(self):
-        # GH 8371
-        # odd results when rounding is needed
 
-        data = """date,time,value
-11-08-2014,00:00:01.093,1
-11-08-2014,00:00:02.159,1
-11-08-2014,00:00:02.667,1
-11-08-2014,00:00:03.175,1
-11-08-2014,00:00:07.058,1
-11-08-2014,00:00:07.362,1
-11-08-2014,00:00:08.324,1
-11-08-2014,00:00:08.830,1
-11-08-2014,00:00:08.982,1
-11-08-2014,00:00:09.815,1
-11-08-2014,00:00:10.540,1
-11-08-2014,00:00:11.061,1
-11-08-2014,00:00:11.617,1
-11-08-2014,00:00:13.607,1
-11-08-2014,00:00:14.535,1
-11-08-2014,00:00:15.525,1
-11-08-2014,00:00:17.960,1
-11-08-2014,00:00:20.674,1
-11-08-2014,00:00:21.191,1"""
+def test_asfreq_non_unique():
+    # GH #1077
+    rng = date_range('1/1/2000', '2/29/2000')
+    rng2 = rng.repeat(2).values
+    ts = Series(np.random.randn(len(rng2)), index=rng2)
 
-        from pandas.compat import StringIO
-        df = pd.read_csv(StringIO(data), parse_dates={'timestamp': [
-            'date', 'time']}, index_col='timestamp')
-        df.index.name = None
-        result = df.resample('6s').sum()
-        expected = DataFrame({'value': [
-            4, 9, 4, 2
-        ]}, index=date_range('2014-11-08', freq='6s', periods=4))
-        assert_frame_equal(result, expected)
+    pytest.raises(Exception, ts.asfreq, 'B')
```

Comment:
btw ok to change these to the new format (if you want; we eventually want to fix all of these), e.g.
```
with pytest.raises(Exception):
    ts.asfreq('B')
```